tests/phpunit/api/v3/ACLPermissionTest.php

```diff
@@ -83,6 +83,22 @@ public function testContactGetNoResultsHook() {
   }
 
   /**
+   * Function tests that an empty where hook returns no results with edit my contact.
+   *
+   * CRM-16512 caused contacts with Edit my contact to be able to view all records.
+   */
+  public function testContactGetNoResultsHookWithEditMyContact() {
+    $this->createLoggedInUser();
+    $this->hookClass->setHook('civicrm_aclWhereClause', array($this, 'aclWhereHookNoResults'));
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = array('access CiviCRM', 'view my contact');
+    $result = $this->callAPISuccess('contact', 'get', array(
+      'check_permissions' => 1,
+      'return' => 'display_name',
+    ));
+    $this->assertEquals(0, $result['count']);
```

Comment:
Sorry - you might be right about the test - I took another test as a starting point & thought 'view my contact' shouldn't change it. But per your comments that assumption is wrong & it should be 1. Actual return is 3 though so the test does demonstrate the failure although it needs tweaking.

NB - I think the issue with your patch is that the $contactID var in that function is just confusing